packages/analytics/analytics-geo-map/src/components/AnalyticsGeoMap.vue

```diff
@@ -0,0 +1,244 @@
+<template>
+  <div
+    id="mapContainer"
+    ref="mapContainer"
+    class="kong-ui-public-analytics-geo-map"
+  />
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted, watch, computed } from 'vue'
+import { Map, Popup } from 'maplibre-gl'
+import type { PropType } from 'vue'
+import type { ColorSpecification, DataDrivenPropertyValueSpecification, ExpressionSpecification, LngLatLike, MapOptions } from 'maplibre-gl'
+import type { LongLat, MapFeatureCollection } from '../types'
+import type { FeatureCollection, Feature, MultiPolygon } from 'geojson'
+import 'maplibre-gl/dist/maplibre-gl.css'
+
+const props = defineProps({
+  countryMetrics: {
+    type: Object as PropType<Record<string, number>>,
+    required: true,
+  },
+  geoJsonData: {
+    type: Object as PropType<MapFeatureCollection>,
+    required: true,
+  },
+  center: {
+    type: Object as PropType<LongLat>,
+    required: false,
+    default: null,
+  },
+  fitToCountry: {
+    type: String,
+    required: false,
+    default: '',
+  },
+  initialZoom: {
+    type: Number,
+    required: false,
+    default: null,
```

Comment:
if the type is a number, the default should be a number

Reply:
Darn, you beat me to it... Maybe a value closer to zero, for a broader map view.

Reply:
I need a way to tell that a user did not provide an initial zoom. Can I type this as `Number | null` ?

Reply:
I believe so yes

Reply:
done